windows/devices/uwb/include/windows/devices/uwb/UwbDeviceConnector.hxx

```diff
@@ -192,8 +195,23 @@ private:
     void
     OnSessionRangingData(::uwb::protocol::fira::UwbRangingData rangingData);
 
+    /**
+     * @brief Internal function to check if there are callbacks present
+     *
+     * @return true
+     * @return false
+     */
+    bool
+    CallbacksPresent();
+
 private:
+    mutable std::shared_mutex m_eventCallbacksGate;
+
+    mutable std::mutex m_callbacksPresentConditionVariableGate;
+    mutable std::condition_variable m_callbacksPresentConditionVariable;
+
     std::unordered_map<uint32_t, std::weak_ptr<::uwb::UwbRegisteredSessionEventCallbacks>> m_sessionEventCallbacks;
+

```

Comment:
nit: Looks like this newline was added, not sure if intentional

Reply:
I personally like grouping the concurrency related members together.